sympy/functions/combinatorial/numbers.py

```diff
@@ -2028,7 +2028,7 @@ def is_motzkin(n):
              m_1 = 1
              i = 2
              while(1):
-                 next_num = ((2*i + 1)/(i + 2))*m_1 + ((3*i - 3)/(i + 2))*m_0
+                 next_num = int(((2*i + 1)/(i + 2))*m_1 + ((3*i - 3)/(i + 2))*m_0)
```

Comment:
Would it be possible to do all computations with integers? Maybe something like

    ((2*i + 1)*m_1 + (3*i - 3)*m_0) // (i + 2)
    
using integer division.

Reply:
Yes i will make some changes regarding this